src/Sanitizer.php

```diff
@@ -126,8 +132,21 @@ private function sanitize_content( string $content ) {
 			}
 		}
 
-		$content = $this->remove_php_tags( $content );
-		$content = $this->remove_js_tags( $content );
+		if ( $type === 'svg' ) {
+			$allowed_tags = $this->get_allowed_tags_for_svg();
+
+			$xml_content = '';
+
+			if ( strpos( $content, '<?xml' ) === 0 ) {
+				$xml_content = substr( $content, 0, strpos( $content, '?>' ) + 2 );
+			}
+
+			$content = $this->remove_js_tags( $content );
```

Comment:
wp_kses has a bug that when <script> contains code wrapped into `// <![CDATA[` old mosaic trick it incorrectly strips this tag, file:

```
<?xml version="1.0" encoding="utf-8"?><!-- Uploaded to: SVG Repo, www.svgrepo.com, Generator: SVG Repo Mixer Tools -->
<svg
  viewBox="0 0 10 10"
  height="120px"
  width="120px"
  xmlns="http://www.w3.org/2000/svg">
  <circle cx="5" cy="5" r="4" />

  <script>
    // <![CDATA[
    function getColor() {
      const R = Math.round(Math.random() * 255)
        .toString(16)
        .padStart(2, "0");

      const G = Math.round(Math.random() * 255)
        .toString(16)
        .padStart(2, "0");

      const B = Math.round(Math.random() * 255)
        .toString(16)
        .padStart(2, "0");

      return `#${R}${G}${B}`;
    }

    document.querySelector("circle").addEventListener("click", (e) => {
      e.target.style.fill = getColor();
    });
    // ]]>
  </script>
</svg>
```

will become 
```
<!-- Uploaded to: SVG Repo, www.svgrepo.com, Generator: SVG Repo Mixer Tools -->
<svg viewBox="0 0 10 10" height="120px" width="120px" xmlns="http://www.w3.org/2000/svg">
  <circle cx="5" cy="5" r="4" />

  
    //  {
      e.target.style.fill = getColor();
    });
    // ]]&gt;
  
</svg>
```
I am keeping this regex method as a last (or more accurate: first) straw to try not to have svg images broken after the upload.